src/main/java/org/folio/services/consortium/ShadowInstanceSynchronizationHandler.java

```diff
@@ -0,0 +1,185 @@
+    } catch (Exception e) {
+      LOG.warn("handle:: Error while handling event for shadow instances synchronization", e);
+      return Future.failedFuture(e);
+    }
+  }
+
+  private boolean isCentralTenantId(String tenantId, ConsortiumData consortiumData) {
+    return tenantId.equals(consortiumData.getCentralTenantId());
+  }
+
+  private Future<Void> synchronizeShadowInstances(DomainEvent<Instance> event, String instanceId,
+                                                  ConsortiumData consortiumData, Map<String, String> headers) {
+    return getShadowInstancesTenantIds(consortiumData.getConsortiumId(), consortiumData.getCentralTenantId(),
+      instanceId, headers)
+      .compose(tenantIds -> updateShadowInstances(event, tenantIds, headers));
+  }
+
+  private Future<List<String>> getShadowInstancesTenantIds(String consortiumId, String centralTenantId,
+                                                           String instanceId, Map<String, String> headers) {
+    String okapiUrl = headers.get(URL);
+    String preparedPath = format(SHARING_INSTANCES_PATH, consortiumId, instanceId);
+    WebClient client = WebClient.wrap(httpClient);
+    HttpRequest<Buffer> request = client.requestAbs(GET, okapiUrl + preparedPath);
+    headers.forEach(request::putHeader);
+    request.addQueryParam(LIMIT_QUERY_PARAM, TENANT_IDS_LIMIT);
+
+    return request.send().compose(response -> {
+      if (response.statusCode() != HTTP_OK) {
+        String msg = format("Error retrieving shadow instances tenant ids, response status: '%s', response body: '%s'",
+          response.statusCode(), response.bodyAsString());
+        LOG.warn("getShadowInstancesTenantIds:: {}", msg);
+        return Future.failedFuture(msg);
+      }
+
+      List<String> affiliationsTenantIds = response.bodyAsJsonObject().getJsonArray(SHARING_INSTANCES_FIELD)
+        .stream()
+        .map(JsonObject.class::cast)
+        .flatMap(sharing ->
+          Stream.of(sharing.getString(TARGET_TENANT_ID_FIELD), sharing.getString(SOURCE_TENANT_ID_FIELD)))
+        .filter(tenantId -> !tenantId.equals(centralTenantId))
+        .toList();
+      return Future.succeededFuture(affiliationsTenantIds);
+    });
+  }
+
+  private Future<Void> updateShadowInstances(DomainEvent<Instance> event, List<String> tenantIds,
+                                             Map<String, String> headers) {
+    LOG.info("updateShadowInstances:: Trying to update shadow instances in the following tenants: {} ", tenantIds);
+    Instance instance = JsonObject.mapFrom(event.getNewEntity()).mapTo(Instance.class);
+    prepareInstanceForUpdate(instance);
+    List<List<String>> tenantsChunks = Lists.partition(tenantIds, instancesParallelUpdatesLimit);
+
+    Future<CompositeFuture> future = Future.succeededFuture();
+    for (List<String> tenantsChunk : tenantsChunks) {
+      future = future.compose(v -> updateShadowInstances(tenantsChunk, instance, headers));
```

Comment:
if one update to a shadow instance fails, it will abort the chain. Update `updateShadowInstances` to prevent this.

Reply:
updated

Reply:
it will work in this way as well because .join in the updateShadowInstances waits for the completion of all single instance updates